packages/init/src/SliceMachineInitProcess.ts

```diff
@@ -606,12 +631,20 @@ Continue with next steps in Slice Machine.
 		});
 	}
 
-	protected async waitingForLogin(): Promise<void> {
-		const { port, url } = await this.manager.user.getLoginSessionInfo();
+	protected async waitingForLogin(
+		onListenCallback?: (
+			sessionInfo: Awaited<
+				ReturnType<typeof this.manager.user.getLoginSessionInfo>
+			>,
+		) => void,
+	): Promise<void> {
+		const sessionInfo = await this.manager.user.getLoginSessionInfo();
+		const { port, url } = sessionInfo;
 		await this.manager.user.nodeLoginSession({
 			port,
 			onListenCallback() {
 				open(url);
+				onListenCallback?.(sessionInfo);
```

Comment:
Probably a question for a more long-term team member, but i wonder why we use a callback instead of the promise we already have here.

Reply:
I suspect that, for this particular case, is because we want to open the browser once the we start listening for auth events, but we only resolve the promise once the authentication succeeded.